dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/MetricsDataS3Client.java

```diff
@@ -0,0 +1,166 @@
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+public class MetricsDataS3Client {
+    private static final Logger LOG = LoggerFactory.getLogger(MetricsDataS3Client.class);
+    private final S3Client s3;
+    private final String bucketName;
+
+    public MetricsDataS3Client(String bucketName) throws URISyntaxException {
+        this.bucketName = bucketName;
+        //TODO should not need to hardcode region since buckets are global, but http://opensourceforgeeks.blogspot.com/2018/07/how-to-fix-unable-to-find-region-via.html
+        this.s3 = S3Client.builder().region(Region.US_EAST_1).build();
+    }
+
+    /**
+     * This constructor is only used for testing with the localstack endpoint
+     * @param bucketName
+     * @param endpointOverride
+     */
+    public MetricsDataS3Client(String bucketName, URI endpointOverride) {
+        this.bucketName = bucketName; // TODO:
+        //TODO should not need to hardcode region since buckets are global, but http://opensourceforgeeks.blogspot.com/2018/07/how-to-fix-unable-to-find-region-via.html
+        this.s3 = S3Client.builder().region(Region.US_EAST_1).endpointOverride(endpointOverride).build();
+    }
+
+    /**
+     * Creates an S3 object containing metrics data for the given GA4GH tool and tool version
+     *
+     * @param toolId The GA4GH Tool ID
+     * @param versionName The GA4GH ToolVersion name
+     * @param platform The platform that the metrics data is from
+     * @param fileName The file name to use. Should be the time that the data was submitted in milliseconds since epoch appended with '.json'
+     * @param ownerUserId The user id of the owner (user that sent the metrics data)
+     * @param metricsData The metrics data in JSON format
+     */
+    public void createS3Object(String toolId, String versionName, String platform, String fileName, long ownerUserId, String metricsData) {
```

Comment:
I'm wondering if it's worth having a "description" or something like that.

Thinking of a use case that Denis once mentioned, where some institute might give us a spreadsheet, and then a curator converts that into a request for this endpoint. It might be nice to have a little more context than just the userid, e.g., "Converted spreadsheet from Genomes R Us covering dates Jan-March".